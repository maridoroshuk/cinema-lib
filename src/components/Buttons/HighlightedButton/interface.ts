import { ReactNode } from 'react';

export interface IUnderlineButton {
  children: ReactNode;
  handleButtonClick: () => void;
  value?: string;
}
