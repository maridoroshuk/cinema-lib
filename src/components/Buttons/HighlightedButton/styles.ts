import styled from 'styled-components';

export const ButtonStyled = styled.button`
  font-weight: ${({ theme }) => theme.sizes.fontWeight[300]};
  font-size: ${({ theme }) => theme.sizes.fontSize[24]}px;
  line-height: ${({ theme }) => theme.sizes.lineHeight[54]}px;
  color: ${({ theme }) => theme.colors.neutral000};
  font-family: ${({ theme }) => theme.fontFamily.Poppins};
  background-color: transparent;
  outline: none;
  border: none;
  cursor: pointer;

  &:hover {
    color: ${({ theme }) => theme.colors.gold};
  }
`;
