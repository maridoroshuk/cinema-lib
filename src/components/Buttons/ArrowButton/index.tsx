import React from 'react';

import ArrowImage from '@/assets/icons/arrow.png';

import { Arrow, Wrapper } from './styles';
import { IArrowButton } from './interface';

export const ArrowButton = ({ onClick }: IArrowButton) => {
  return (
    <Wrapper onClick={onClick}>
      <Arrow src={ArrowImage} alt="arrow (hide)" />
    </Wrapper>
  );
};
