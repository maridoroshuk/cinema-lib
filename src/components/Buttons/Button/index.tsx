import React from 'react';

import { ButtonStyled } from '@/components/Buttons/Button/styles';

import { IButtonProps } from './interface';

export const Button = ({
  children,
  size,
  backgroundColor,
  type,
  value,
  onClick,
  dataCy,
}: IButtonProps) => {
  return (
    <ButtonStyled
      backgroundColor={backgroundColor}
      size={size}
      type={type || 'button'}
      value={value}
      onClick={onClick}
      data-cy={dataCy}
    >
      {children}
    </ButtonStyled>
  );
};
