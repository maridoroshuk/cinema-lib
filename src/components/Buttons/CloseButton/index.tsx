import React from 'react';
import { ICloseButtonProps } from './interface';
import { ButtonWrapper, StyledButton } from './styles';

export const CloseButton = ({ onClose }: ICloseButtonProps) => {
  return (
    <ButtonWrapper>
      <StyledButton onClick={onClose} />
    </ButtonWrapper>
  );
};
