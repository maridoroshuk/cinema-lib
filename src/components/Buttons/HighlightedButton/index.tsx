import React from 'react';
import { IUnderlineButton } from './interface';
import { ButtonStyled } from './styles';

export const HighlightedButton = ({
  children,
  handleButtonClick,
  value,
}: IUnderlineButton) => {
  return (
    <ButtonStyled onClick={handleButtonClick} value={value || ''}>
      {children}
    </ButtonStyled>
  );
};
