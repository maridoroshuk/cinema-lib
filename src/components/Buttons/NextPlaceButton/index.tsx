import React from 'react';

import NextPlaceButton from '@/assets/icons/next_place_button.png';

import { Next, Wrapper } from './styles';

export interface IArrowButton {
  onClick?: () => void;
}

export const NextPlace = ({ onClick }: IArrowButton) => {
  return (
    <Wrapper onClick={onClick}>
      <Next src={NextPlaceButton} />
    </Wrapper>
  );
};
