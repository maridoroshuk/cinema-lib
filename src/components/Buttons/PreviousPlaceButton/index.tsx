import React from 'react';

import PreviousPlaceButton from '@/assets/icons/previous_place_button.png';

import { Previous, Wrapper } from './styles';

export interface IArrowButton {
  onClick?: () => void;
}

export const PreviousPlace = ({ onClick }: IArrowButton) => {
  return (
    <Wrapper onClick={onClick}>
      <Previous src={PreviousPlaceButton} />
    </Wrapper>
  );
};
