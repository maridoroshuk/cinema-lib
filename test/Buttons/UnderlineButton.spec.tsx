import React from 'react';
import { render } from '@testing-library/react';

import { HighlightedButton } from '@/components/Buttons/HighlightedButton';

import 'jest-styled-components';

import { setupTestProvider } from '../setupTests';

describe('UnderlineButton', () => {
  it('renders', () => {
    const handleClick = jest.fn();
    const { container } = render(
      setupTestProvider(
        <HighlightedButton handleButtonClick={handleClick}>
          Settings
        </HighlightedButton>,
      ),
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
