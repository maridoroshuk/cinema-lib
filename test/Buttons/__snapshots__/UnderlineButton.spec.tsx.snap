// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnderlineButton renders 1`] = `
.c0 {
  font-weight: 300;
  font-size: 24px;
  line-height: 54px;
  color: #FFFFFF;
  font-family: 'Poppins',sans-serif;
  background-color: transparent;
  outline: none;
  border: none;
  cursor: pointer;
}

.c0:hover {
  color: #D98639;
}

<button
  class="c0"
  value=""
>
  Settings
</button>
`;
